src/components/steps/nameAndDescription/nameAndDescription.tsx

```diff
@@ -12,10 +14,54 @@ const defaultNameAndDescription = {
   name: '',
 };
 
-const NameAndDescription = () => {
+export type StepsProps = {
+  setSubmitStep: React.Dispatch<
+    React.SetStateAction<{
+      initialized: boolean;
+      submitted: boolean;
+    }>
+  >;
+
+  submitStep: {
+    initialized: boolean;
+    submitted: boolean;
+  };
+};
+
+const NameAndDescription: React.FC<StepsProps> = ({
+  submitStep,
+  setSubmitStep,
+}) => {
   const [nameAndDescription, setNameAndDescription] =
     useState<NameAndDescriptionState>(defaultNameAndDescription);
 
+  const { recipeToUpload, setRecipeToUpload, setDisplayMessage } =
+    useContext(StorageContext);
+
+  useEffect(() => {
+    if (recipeToUpload?.description && recipeToUpload.name) {
+      const { description, name } = recipeToUpload;
+      setNameAndDescription({ description, name });
+    }
+  }, []);
+
+  useEffect(() => {
+    if (submitStep.initialized) {
+      try {
+        const { description, name } = nameAndDescription;
+        if (!description || !name) {
+          setSubmitStep({ ...submitStep, initialized: false });
+          throw new Error('All fields are required.');
+        }
+
+        setRecipeToUpload({ ...recipeToUpload, description, name });
+        setSubmitStep({ initialized: false, submitted: true });
+      } catch (error) {
+        setDisplayMessage((error as Error).message);
+      }
+    }
+  }, [submitStep.initialized]);
```

Comment:
nameAndDescription should be added to the list of dependencies. Otherwise you risk using an out of sync value